### DISCUSSION TITLE
SetPlatform for the new Arm7

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3218] [Extension Q&A] [answered]
I have this code below and Cake.CoreCLR 0.38.4 and I need to set the iPhone configuration for ARM7.

Can this be added? Will using WithProperty work also? I believe this is the new architecture for iOS builds.

I'm having issues with switching the CodesignKey using MSBuild

I looked here and did not see a setting for it: http://develop.cakebuild.net/api/Cake.Common.Tools.MSBuild/PlatformTarget/ 

Also, WithProperty does work, since iPhone is generally used in peoples projects. It would be nice to add an enum or exend the SetPlatform to allow a string.

```MSBuild(BuildParameters.SolutionFile.ToString(), configurator =>
			configurator
				.SetConfiguration(BuildParameters.Configuration)
				.SetPlatformTarget(BuildParameters.Platform)
				.SetVerbosity(Verbosity.Diagnostic)
				.WithProperty("BuildIpa", "true")
				.WithProperty("CodesignKey",  BuildParameters.CertTypes[BranchProject].IOSTeam)
				.WithProperty("CodesignProvision", provisionType) 
				.WithProperty("MtouchNoSymbolStrip", "false")
				.WithProperty("ArchiveOnBuild", "true")
				.WithLogger(
					Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
					"XmlFileLogger",
					$"logfile=\"{BuildParameters.LogPath}\";verbosity=Detailed;encoding=UTF-8"
					)
				);```

**Comment by augustoproiete** [comment, marked as answer]
Assuming `ARM7` is a more recent platform target, then it's not part of the [enum](http://cakebuild.net/api/Cake.Common.Tools.MSBuild/PlatformTarget/) yet (should be added in a future version of Cake), so in the meantime you should use `.WithProperty("Platform", "xxxx")`

e.g.

```diff
configurator
    .SetConfiguration(BuildParameters.Configuration)
-    .SetPlatformTarget(BuildParameters.Platform)
+    .WithProperty("Platform", "ARM7")
    .SetVerbosity(Verbosity.Diagnostic)
    .WithProperty("BuildIpa", "true")
    .WithProperty("CodesignKey",  BuildParameters.CertTypes[BranchProject].IOSTeam)
    .WithProperty("CodesignProvision", provisionType) 
    .WithProperty("MtouchNoSymbolStrip", "false")
    .WithProperty("ArchiveOnBuild", "true")
    .WithLogger(
        Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
        "XmlFileLogger",
        $"logfile=\"{BuildParameters.LogPath}\";verbosity=Detailed;encoding=UTF-8"
        )
    );
```

It would be great if you could point us to documentation where we can see which platform targets we should be adding to the enumeration (PRs welcome as well).

**Reply by nick5454** [reply]
Here is a link I found: https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/compiling-for-different-devices?tabs=windows looks like they update this. Will post more if I find any @augustoproiete

**Reply by nick5454** [reply]
I can start PR's, but swamped with pipelines for the next few days.

**Reply by augustoproiete** [reply]
Thanks!

Looks like at the very least we should be adding:

- `ARMv6`
- `ARMv7`
- `ARMv7s`

Right?

Did you mean `ARMv7` instead of `ARM7` or is `ARM7` something new?

**Reply by nick5454** [reply]
@augustoproiete I think I am wrong, saw it in a post somewhere. I would go with the documentation and I have a 5 apps on the pipeline, so I can test those

**Reply by augustoproiete** [reply]
Tracking here for the known ones: https://github.com/cake-build/cake/issues/3222
